secret.go

```diff
@@ -0,0 +1,22 @@
+package promconfig
+
+const secretToken = "<secret>"
+
+// Secret special type for storing secrets.
+type Secret string
+
+// MarshalYAML implements the yaml.Marshaler interface for Secrets.
+func (s Secret) MarshalYAML() (interface{}, error) {
```

Comment:
@BupycHuk @artemgavrilov I changed logic here a little bit and remove logic that masks values by default. In that case, these changes will break everything for PMM at least because it does config management there.

I restored the old code and changed it a little bit. In that case, we have yaml.Marshal and yaml.Unmarshal that does not change masked values and config.Mask() to mask everything.

I thought about copying the struct and modifying values only on the copied struct, but it seems a little bit redundant because of the following reasons

1.  People usually work with configs by doing Unmarshal, changing values, and marshal again. We don't have any syntax sugar as upstream config have
2. Mask method modifies values of unmarshaled struct in that case it can lead to some misuse in the future. Right now we will use it only for masking sensitive data on /logs.zip endpoint. I think we can add copying of struct a little bit later
What do you think?

Reply:
If we don't have custom `Marshal/Unmarshal` functions then it seems it has no sense to have custom `Secret` type for that. Changing field type is breaking change, we should release it like `v2` and use should update his sources. If we plan to use `Mask` function that replaces sensitive data then tag `secret:true` looks better. It will be a minor release and will not affect anyone.